oid4vc/oid4vc/models/dcql_query.py

```diff
@@ -0,0 +1,412 @@
+    )
+    namespace = fields.Str(
+        required=False,
+        metadata={
+            "description": "",  # TODO: Add description
+        },
+    )
+    claim_name = fields.Str(
+        required=False,
+        metadata={
+            "description": "",  # TODO: Add description
+        },
+    )
+    path = fields.List(
+        fields.Raw,
+        required=False,
+        metadata={
+            "description": "",  # TODO: Add description
+        },
+    )
+    values = fields.List(
+        fields.Raw,
+        required=False,
+        metadata={
+            "description": "Values of the claims query.",
+        },
+    )
+
+    @validates_schema
+    def validate_fields(self, data, **kwargs):
+        """Validate schema fields."""
+        namespace = data.get("namespace")
+        claim_name = data.get("claim_name")
+
+        if namespace and not claim_name:
+            raise ValidationError("Cannot have namespace without claim_name.")
+        if claim_name and not namespace:
+            raise ValidationError("Cannot have claim_name without namespace.")
+
+        path = data.get("path")
+        if path and namespace or path and claim_name:
+            raise ValidationError("Cannot have path and namespace or claim_name.")
+
+        if path:
+            for p in path:
+                if not (
+                    isinstance(p, str) or isinstance(p, int) or p is None
+                ):  # TODO: check logic here
+                    raise ValidationError("Path elements must be string, int, or None.")
+
+        values = data.get("values")
+        if values:
+            for v in values:
+                if not (
+                    isinstance(v, str) or isinstance(v, int) or isinstance(v, bool)
+                ):  # TODO: check logic here
+                    raise ValidationError(
+                        "Values elements must be string, int, or bool."
+                    )
+
+
+ClaimQueryID = str
+
+
+class CredentialMeta(BaseModel):
+    """Metadata for credential query."""
+
+    class Meta:
+        """Credential Meta Schema Metadata."""
+
+        schema_class = "CredentialMetaSchema"
+
+    def __init__(
+        self,
+        query_type: Optional[str] = None,
+        doctype_values: Optional[List[str]] = None,
+        vct_values: Optional[List[str]] = None,
+    ):
+        """Initialize a CredentialMeta model."""
+        super().__init__()
+
+        self.query_type = query_type
+        self.doctype_values = doctype_values
+        self.vct_values = vct_values
+
+
+class CredentialMetaSchema(BaseModelSchema):
+    """Schema for credential query metadata."""
+
+    class Meta:
+        """Credential Meta Schema Metadata."""
+
+        model_class = "CredentialMeta"
+
+    query_type = fields.Str(
+        required=False,
+        validate=OneOf(
+            [
+                "SdJwtVcMeta",
+                "MdocMeta",
+            ]
+        ),
+    )
```

Comment:
Does this mean that the user needs to provide a `query_type` property in the `meta` of their DCQL query in order for it to parse correctly?

Reply:
I don't think so? I don't know enough about Marshmallow to say for sure, but I think the `required=False` means that it doesn't need to be present, but if it is, it needs to be one of the list.
That whole thing might not be useful to us at all; I had thought that it would be a helpful way to quickly determine the query type for some other things, but I didn't end up populating it anywhere or using it in the other places where validation happens, so it might be better to cut it

Reply:
Yeah, let's cut it since it's not technically in the spec either